### ISSUE TITLE
Invalid Swift Support

### ISSUE THREAD

**Issue by yn5** [fastlane/fastlane#6341]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description
I'm trying to build a beta version for my app for TestFlight but I get an email from Apple telling me the SwiftSupport folder is missing:

```
Dear developer,

We have discovered one or more issues with your recent delivery for "Your App". To process your delivery, the following issues must be corrected:

Invalid Swift Support - The SwiftSupport folder is missing. Rebuild your app using the current public (GM) version of Xcode and resubmit it.

Regards,

The App Store team
```

### Configuration Files

Fastfile:
```ruby
lane :beta do

    match(type: "appstore")

    increment_build_number(
      xcodeproj: "./ios/yourapp.xcodeproj"
    )

    gym(
      scheme: "yourapp",
      project: "./ios/yourapp.xcodeproj",
      use_legacy_build_api: true
    )


    pilot(
      username: "mymail@gmail.com"
    )

    sh "pilot list -u mymail@gmail.com"
  end
```

### Environment

fastlane 1.104.0

Do you use bundler to execute fastlane (i.e. `bundle exec fastlane`)?
No

Do you use a Ruby environment manager (e.g. `chruby`, `rbenv`, `rvm`)?
No

**Comment by TKBurner** [comment]
@yn5 Hope you are doing well. Did you run match development prior to running match appstore? Thanks!

**Comment by KrauseFx** [comment]
> Rebuild your app using the current public (GM) version of Xcode and resubmit it.

This seems to be due to the used Xcode version, make sure to use the latest public Xcode release to submit your app

**Comment by yn5** [comment]
@KrauseFx Thanks for your suggestion, I'm running Xcode 8.0.

@TKBurner I'm doing well thanks :)

I think I ran `match development` before. Tried it again and got the following error doing so:
```
[01:28:50]: An error occured while verifying your certificates and profiles with the Apple Developer Portal.
[01:28:50]: If you already have your certificates stored in git, you can run `match` in readonly mode
[01:28:50]: to just install the certificates and profiles without accessing the Dev Portal.
[01:28:50]: To do so, just pass `readonly: true` to your match call.
[01:28:50]: 🔒  Successfully encrypted certificates repo

Looking for related GitHub issues on fastlane/fastlane...

➡️  Crash when there are no devices on the portal
   https://github.com/fastlane/fastlane/issues/2004 [closed] 9 💬
   7 weeks ago


[!] Apple provided the following error info:
	There were errors in the data supplied. Please correct and re-submit.
	There are no current devices on this team matching the provided device IDs.
```
I've tried removing all certificates from the developer portal and used a clean repository but unfortunately no luck.

When running `match development` twice without removing certificates from the Portal in between I got the error:
`[!] Could not create another certificate, reached the maximum number of available certificates.`

Then I tried removing a few certificates from my keychain after Googling around for the first error and I'm stuck with the following error now(, for which I guess deserves a new github issue if I don't manage to figure it out myself):
```
[02:08:31]: Cloning remote git repo...
Cloning into '/var/folders/lj/qdby8x0x5l52mpn5_hg9ts000000gn/T/d20161005-62827-sz5d8q'...
fatal: could not read Password for 'https://myaccount@bitbucket.org': terminal prompts disabled
[02:08:33]: Exit status: 128
[02:08:33]: 🔒  Successfully encrypted certificates repo

[!] Exit status: 128
```

**Comment by KrauseFx** [comment]
Easiest fix is to run `match nuke` to get a clean start 👍

**Comment by yn5** [comment]
@KrauseFx Match nuke unfortunately also results in the error:

```
[02:08:31]: Cloning remote git repo...
Cloning into '/var/folders/lj/qdby8x0x5l52mpn5_hg9ts000000gn/T/d20161005-62827-sz5d8q'...
fatal: could not read Password for 'https://myaccount@bitbucket.org': terminal prompts disabled
[02:08:33]: Exit status: 128
[02:08:33]: 🔒  Successfully encrypted certificates repo

[!] Exit status: 128
```

**Comment by KrauseFx** [comment]
You didn't properly set up git, run `--verbose` and make sure the `git clone` command succeeds

**Comment by asfalcone** [comment]
@yn5 Were you able to get git setup correctly so that _match_ is working for you now? We'd like to close this issue if it was an environment problem.

**Comment by yn5** [comment]
@asfalcone Unfortunately I did not have time to work on the project anymore but the last time I did the problem was still there. Is it ok to leave the ticket open a little while longer?

**Comment by TKBurner** [comment]
@yn5 Definitely! Let us know when you have an update.

**Comment by yn5** [comment]
@TKBurner @KrauseFx Git is successfully cloning again, I can run `match nuke` and all other commands without any problems but, I'm back to square one (the exact problem described in the opening post).

The order of commands I'm executing now is:
```
match nuke development;
match nuke distribution;
match development;
match appstore;
fastlane beta;
```

Output from the last `pilot` step: 

```
[13:47:34]: -------------------
[13:47:34]: --- Step: pilot ---
[13:47:34]: -------------------
[13:47:34]: Login to iTunes Connect (me@gmail.com)
Multiple iTunes Connect teams found, please enter the number of the team you want to use:
Note: to automatically choose the team, provide either the iTunes Connect Team ID, or the Team Name in your fastlane/Appfile:

  itc_team_id "00012221"

or

  itc_team_name "Studio"

1) "Studio" (118406010)
2) "Me" (117930180)
1
[13:47:48]: Login successful
[13:47:48]: Ready to upload new build to TestFlight (App: 1232341)...
[13:47:48]: Going to upload updated app to iTunes Connect
[13:47:48]: This might take a few minutes. Please don't interrupt the script.
[13:48:50]: iTunes Transporter successfully finished its job
[13:48:50]: ------------------------------------------------------------------------------------------------------
[13:48:50]: Successfully uploaded package to iTunes Connect. It might take a few minutes until it's visible online.
[13:48:50]: ------------------------------------------------------------------------------------------------------
[13:48:50]: Successfully uploaded the new binary to iTunes Connect
[13:48:50]: If you want to skip waiting for the processing to be finished, use the `skip_waiting_for_build_processing` option
[13:48:50]: Waiting for iTunes Connect to process the new build
[13:49:21]: Variable Dump:
[13:49:21]: {:DEFAULT_PLATFORM=>:ios, :PLATFORM_NAME=>:ios, :LANE_NAME=>"ios beta", :SIGH_PROFILE_TYPE=>"app-store", :BUILD_NUMBER=>"3", :IPA_OUTPUT_PATH=>"/Path/to/app.ipa", :DSYM_OUTPUT_PATH=>"/Path/to/dogwalk.app.dSYM.zip", :XCODEBUILD_ARCHIVE=>"/Path/to/app 2016-10-19 13.46.34.xcarchive"}
[13:49:21]: Error receiving the newly uploaded binary, please check iTunes Connect

+------+-------------------------------------+-------------+
|                     fastlane summary                     |
+------+-------------------------------------+-------------+
| Step | Action                              | Time (in s) |
+------+-------------------------------------+-------------+
| 1    | Verifying required fastlane version | 0           |
| 2    | default_platform                    | 0           |
| 3    | match                               | 10          |
| 4    | increment_build_number              | 0           |
| 5    | gym                                 | 62          |
| 💥    | pilot                               | 106         |
+------+-------------------------------------+-------------+

[13:49:21]: fastlane finished with errors

[!] Error receiving the newly uploaded binary, please check iTunes Connect
```

**Comment by hjanuschka** [comment]
have you tried disabling `use_legacy_api` ? - and maybe you need to set `toolchain: :swift_2_3` on gym.

**Comment by yn5** [comment]
@hjanuschka Thanks for your help!
I've tried the following gym configs without any luck, the problem stays exactly the same as in the opening post:

```
gym(
      scheme: "dogwalk",
      project: "./ios/dogwalk.xcodeproj"
    )
```
```
gym(
      scheme: "dogwalk",
      project: "./ios/dogwalk.xcodeproj",
      toolchain: :swift_2_3
    )
```
```
gym(
      scheme: "dogwalk",
      project: "./ios/dogwalk.xcodeproj",
      toolchain: :swift_2_3,
      use_legacy_build_api: true
    )
```

**Comment by hjanuschka** [comment]
can you give  
http://stackoverflow.com/a/38930579 a try
--  
Helmut Januschka
Sent with Airmail

**Comment by yn5** [comment]
@hjanuschka Thanks for chiming in! This is all I see when I search the target's Build Settings for embed (I'm on Xcode 8.0):

<img width="1400" alt="screen shot 2016-10-19 at 17 02 17" src="https://cloud.githubusercontent.com/assets/1705585/19524419/f02a383e-961d-11e6-822d-28a5ccb7103c.png">

**Comment by TKBurner** [comment]
Hi @yn5 can you make sure you have the most recent version of xcode command line tools installed?

**Comment by yn5** [comment]
@TKBurner 
```
$ xcode-select -v
xcode-select version 2343.
```
And the App Store does not have any updates available.

@hjanuschka @TKBurner @KrauseFx Just noticed I never mentioned the app we're discussing is a [react-native](https://facebook.github.io/react-native/) app. In case that could help figuring out what is going on here.

**Comment by hjanuschka** [comment]
i have `2345.1`  - so maybe you should remove xcode and reinstall it - from appstore.
or use  xcode-install -> https://github.com/KrauseFx/xcode-install

**Comment by yn5** [comment]
@hjanuschka I've tried reinstalling Xcode trough the app store as well as with the latest version (2.0.7) of xcode-install. But `xcode-select -v` keeps returning 2343. Are you sure you are on OSX 10.11(.6) too?

**Comment by hjanuschka** [comment]
i am on 10.12 actually, have you tried reinstalling the cli tools: `xcversion install-cli-tools`

**Comment by yn5** [comment]
@hjanuschka Yes I've tried that too.

**Comment by TKBurner** [comment]
@yn5 Can you reproduce this on a blank project? I want to see what we can do to help. Thanks!

**Comment by yn5** [comment]
@TKBurner @KrauseFx @hjanuschka @asfalcone I indeed concluded that it must be my code base. I'm currently rebuilding the project from scratch and releasing in between every addition in order to figure out what the problem is. And I have indeed managed to get a very rudimentary version in TestFlight.

Thanks all a million for your help! I think the issue can be closed, unless you'd like me to update you all on the process here, that's fine with me too.
